lib/Controller/ControllerExecutor.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Website\Controller;
+
+use Doctrine\Website\Builder\SourceFile;
+use ReflectionClass;
+use RuntimeException;
+use function sprintf;
+
+class ControllerExecutor
+{
+    /** @var ControllerFactory */
+    private $controllerFactory;
+
+    public function __construct(ControllerFactory $controllerFactory)
+    {
+        $this->controllerFactory = $controllerFactory;
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function execute(SourceFile $sourceFile) : array
+    {
+        [$className, $methodName] = $sourceFile->getParameter('controller');
+
+        $controller = $this->controllerFactory->create($className);
+
+        $reflectionMethod = (new ReflectionClass($controller))->getMethod($methodName);
+
+        $controllerResult = $reflectionMethod->invokeArgs($controller, [$sourceFile]);
```

Comment:
Only 3 of the currently 9 controller actions need `$sourceFile` injected. In the other controller actions it is an unused argument. I'm not sure about that approach. Why did you choose the way of ControllerFactory, ControllerExecutor, ControllerResult and controller methods having the need to get the same `SourceFile` argument?